### PR TITLE
Ports L2 Module To bm_core

### PR DESCRIPTION
**Pull request by matt001k**
Refactors code to utilize bm_core API (bm_os, etc.)
Abstracts lwip elements from l2 file and provides a generic API that can be used across multiple network stacks

### CODE REVIEW

**Comment by matt001k** [comment]
I have not moved the L2 file yet to bm_core. I plan on opening another PR after this just to migrate it over. That way y'all can see the diff of what I changed in this PR.

File: src/lib/bcmp/bm/bm_l2.cpp [hunk 1]
```diff
@@ -1,524 +1,510 @@
...
 #include "eth_adin2111.h"
 #include "lwip/ethip6.h"
 #include "lwip/prot/ethernet.h"
 #include "lwip/snmp.h"
 #include "task_priorities.h"
+extern "C" {
+#include "bm_ip.h"
+#include "bm_os.h"
+#include "util.h"
+}
```
Comment (matt001k) [inline src/lib/bcmp/bm/bm_l2.cpp:9]: These will go away when migrated to bm_core repo.

File: src/lib/bcmp/bm/bristlemouth.h [hunk 2]
```diff
@@ -1,21 +1,8 @@
 #pragma once
 
+#include "serial.h"
```
Comment (victorsowa12) [inline src/lib/bcmp/bm/bristlemouth.h]: why is serial.h needed here?

File: src/lib/bcmp/bm/bristlemouth.cpp [hunk 3]
```diff
@@ -126,6 +82,8 @@ void bcl_init(void) {
   \return pointer to string with ip address
 */
 const char *bcl_get_ip_str(uint8_t idx) {
-  configASSERT(idx < LWIP_IPV6_NUM_ADDRESSES);
-  return (ip6addr_ntoa(netif_ip6_addr(&netif, idx)));
+  (void)idx;
+  //configASSERT(idx < LWIP_IPV6_NUM_ADDRESSES);
+  //return (ip6addr_ntoa(netif_ip6_addr(&netif, idx)));
+  return "none\n";
```
Comment (victorsowa12) [inline src/lib/bcmp/bm/bristlemouth.cpp]: Is it possible to also move this over to the IP abstraction? I think it is only used by the CLI command `info` so I don't think it really needs to be a part of bcl.

File: src/lib/bcmp/bm/bristlemouth.h (hunk 2 above)
Comment (matt001k) [reply src/lib/bcmp/bm/bristlemouth.h]: Oops, I really have no idea how this got added :(

File: src/lib/bcmp/bm/bristlemouth.cpp (hunk 3 above)
Comment (matt001k) [reply src/lib/bcmp/bm/bristlemouth.cpp]: Ahhh yes I meant to do that, thanks for the catch.

**Review by victorsowa12 — APPROVED** [review]
🤘

**Pull request merged**